syntax = "proto3";

import "google/protobuf/empty.proto";

service ArchivoService {
  rpc EnviarArchivo(Archivo) returns (Respuesta) {}
  rpc chunksArchivo(google.protobuf.Empty) returns (ListaChunks) {}
  rpc ListarArchivos(google.protobuf.Empty) returns (Lista) {}
  rpc HeartBeat(Nombre) returns (Respuesta) {}
}

message Nombre {
  string nombre = 1;
}

message Archivo{
    repeated bytes contenido = 1;
    string fileName = 2;
    string fileExt = 3;
}
  

message Respuesta {
  string mensaje = 1;
}

message ListaChunks {
  repeated string keys = 1;
  repeated string values = 2;
}

message Lista {
  repeated string archivos = 1;
}